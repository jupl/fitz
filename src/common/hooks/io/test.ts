import {expect, test} from '@jest/globals'
import {renderHook} from '@testing-library/react-hooks'

test('useDecode hook', async () => {
  const IO = await import('io-ts')
  const {useDecode} = await import('.')
  const {result} = renderHook(() => useDecode(IO.number, 1))
  expect(result.current).toMatchSnapshot()
})

test('useEncode hook', async () => {
  const IO = await import('io-ts')
  const {useEncode} = await import('.')
  const {result} = renderHook(() => useEncode(IO.number, 1))
  expect(result.current).toMatchSnapshot()
})
