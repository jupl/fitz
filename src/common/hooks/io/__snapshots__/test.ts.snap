// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useDecode hook 1`] = `1`;

exports[`useEncode hook 1`] = `1`;
